### ISSUE TITLE
Serializing my BigFloat class as a long number (i.e. unquoted) doesn't work.

### ISSUE THREAD

**Issue by ivands** [jawj/json-custom-numbers#1]
For some reason, trying to serialize a bigint works fine.
But when I try to serialize my object, it won't.

Example:
```ts
import { BigFloat } from '@awsless/big-float'
import { stringify } from 'json-custom-numbers'

describe('JSON', () => {
	it('should stringify', () => {
		const one = new BigFloat(1)
		const result = stringify(one, undefined, undefined, (_, value) => {
			if (value instanceof BigFloat) {
				return value.toString()
			}
		})

		// will return "1" instead of 1 (i.e. unquoted)
		expect(result).toBe('1')
	})
})
```

**Comment by jawj** [comment]
Thanks for opening this issue.

I see why this is happening. To match native `JSON.stringify()`, my `stringify()` calls `toJSON()` on objects that offer it, such as `BigFloat`. See https://github.com/jawj/json-custom-numbers/blob/e66231e5a9d502d6b897058429a43361be5d8fb9/src/stringify.ts#L95.

This happens before your custom serializer is called, so that in your example the value becomes a string before it gets passed to your anonymous function.

But I guess that `toJSON` probably shouldn't be called at all when a custom serializer is provided.

**Comment by ivands** [comment]
Or the custom serializer function can be called before `toJSON()`.
This will allow the custom serializer to serialize the objects however it wants and will still fall back to using `toJSON()`.

**Comment by jawj** [comment]
Hmm, I took another look and I don't think that works. The custom function is a serializer, not a replacer: it returns a string that goes straight into the output JSON stream. Strings don't have a `toJSON` method (or not one that changes anything), so calling `toJSON` afterwards doesn't make sense.

I think what I'll do instead is add an option that stops `toJSON` being called on any object during serialization.

**Comment by ivands** [comment]
I get that. But the custom function will only serialize whatever you want.
Anything else won't be.
So, it could still be possible for an object to have a `toJSON` function after the custom serialize function runs.

**Comment by jawj** [comment]
I've published 3.1.0, which has an `options` object with a `skipToJSON` key (documented in the types but not yet the README). I think this should fix your issue.

**Comment by ivands** [comment]
It solved my issue.
Thanks for the fix.
